bpf/aya-bpf/src/maps/xdp/cpu_map.rs

```diff
@@ -0,0 +1,124 @@
+use core::{cell::UnsafeCell, mem};
+
+use aya_bpf_bindings::bindings::bpf_cpumap_val;
+
+use crate::{
+    bindings::{bpf_map_def, bpf_map_type::BPF_MAP_TYPE_CPUMAP},
+    helpers::bpf_redirect_map,
+    maps::PinningType,
+};
+
+/// An array of available CPUs.
+///
+/// XDP programs can use this map to redirect packets to a target CPU for processing.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 4.15.
+///
+/// # Examples
+///
+/// ```rust,no_run
+/// use aya_bpf::{bindings::xdp_action, macros::{map, xdp}, maps::CpuMap, programs::XdpContext};
+///
+/// #[map]
+/// static MAP: CpuMap = CpuMap::with_max_entries(8, 0);
+///
+/// #[xdp]
+/// fn xdp(_ctx: XdpContext) -> i32 {
+///     // Redirect to CPU 7 or drop packet if no entry found.
+///     MAP.redirect(7, xdp_action::XDP_DROP as u64)
+/// }
+/// ```
+#[repr(transparent)]
+pub struct CpuMap {
+    def: UnsafeCell<bpf_map_def>,
+}
+
+unsafe impl Sync for CpuMap {}
+
+impl CpuMap {
+    /// Creates a [`CpuMap`] with a set maximum number of elements.
+    ///
+    /// In a CPU Map, an entry represents a CPU core. Thus there should be as many entries as there
+    /// are CPU cores on the system. It can be set to zero here, and updated by userspace at
```

Comment:
"It" is ambiguous here. `max_entries can be set to zero here ...`

Reply:
done